tools/datagen/src/bin/datagen.rs

```diff
@@ -349,7 +349,23 @@ fn main() -> eyre::Result<()> {
         _ => unreachable!(),
     };
 
-    selected_keys.into_par_iter().try_for_each(|key| {
+    // Split the selected_keys into two sets: segmenter_keys and other_keys.
+    let (segmenter_keys, mut other_keys): (HashSet<ResourceKey>, HashSet<ResourceKey>) =
+        selected_keys
+            .into_iter()
+            .partition(|k| icu_segmenter::ALL_KEYS.contains(k));
+
+    if !segmenter_keys.is_empty() {
+        // Add resource keys for Unicode Properties that the segmenter data depends on.
```

Comment:
Issue: It depends on them only for data building.

What you should do is to load the dependency keys directly from the source provider (icu_provider_uprops) rather than from the postcard file. It is slightly slower, but it doesn't introduce any weird dependencies and doesn't impact the main data file.

Reply:
Praise: I like the idea of using `icu_provider_uprops` to generate segmenter rule data, and then `datagen` can support customized segmenter key from the start.